sdcflows/interfaces/bspline.py

```diff
@@ -288,116 +284,112 @@ def _run_interface(self, runtime):
-            unwarp = B0FieldTransform(
-                coeffs=[nb.load(cname) for cname in self.inputs.in_coeff],
-            )
+        from sdcflows.transform import B0FieldTransform
 
-        if not isdefined(self.inputs.num_threads) or self.inputs.num_threads < 2:
-            # Linear execution (1 core)
-            outputs = [
-                _b0_resampler(
-                    fname,
-                    self.inputs.in_coeff,
-                    pe_dir[i],
-                    ro_time[i],
-                    hmc_mats[i],
-                    unwarp,  # if no HMC matrices, interpolator can be shared
-                    runtime.cwd,
-                )
-                for i, fname in enumerate(self.inputs.in_data)
-            ]
-        else:
-            # Embarrasingly parallel execution
-            from concurrent.futures import ProcessPoolExecutor
-
-            outputs = [None] * len(self.inputs.in_data)
-            with ProcessPoolExecutor(max_workers=self.inputs.num_threads) as ex:
-                outputs = ex.map(
-                    _b0_resampler,
-                    self.inputs.in_data,
-                    [self.inputs.in_coeff] * n,
-                    pe_dir,
-                    ro_time,
-                    hmc_mats,
-                    [None] * n,  # force a new interpolator for each process
-                    [runtime.cwd] * n,
-                )
+        # Pre-cached interpolator object
+        unwarp = B0FieldTransform(coeffs=[nb.load(cname) for cname in self.inputs.in_coeff])
 
-        (
-            self._results["out_corrected"],
-            self._results["out_warp"],
-            self._results["out_field"],
-        ) = zip(*outputs)
+        # We can now write out the fieldmap
+        self._results["out_field"] = fname_presuffix(
+            self.inputs.in_data,
+            suffix="_field",
+            newpath=runtime.cwd,
+        )
 
-        out_fields = set(self._results["out_field"]) - set([None])
-        if len(out_fields) == 1:
-            self._results["out_field"] = out_fields.pop()
+        # HMC matrices are only necessary when reslicing the data (i.e., apply())
+        # Check the length of in_xfms matches that of in_data
+        self._results["out_corrected"] = fname_presuffix(
+            self.inputs.in_data,
+            suffix="_sdc",
+            newpath=runtime.cwd,
+        )
 
+        unwarp.apply(
+            self.inputs.in_data,
+            self.inputs.pe_dir,
+            self.inputs.ro_time,
+            xfms=self.inputs.in_xfms if isdefined(self.inputs.in_xfms) else None,
+            fmap2data_xfm=(
+                None if not isdefined(self.inputs.fmap2data_xfm) else self.inputs.fmap2data_xfm
+            ),
+            approx=self.inputs.approx,
+            num_threads=(
+                None if not isdefined(self.inputs.num_threads) else self.inputs.num_threads
+            ),
+        ).to_filename(self._results["out_corrected"])
```

Comment:
to keep a consistent syntax
```suggestion
        unwarp.apply(
            self.inputs.in_data,
            self.inputs.pe_dir,
            self.inputs.ro_time,
            xfms=self.inputs.in_xfms if isdefined(self.inputs.in_xfms) else None,
            fmap2data_xfm=(
                self.inputs.fmap2data_xfm if isdefined(self.inputs.fmap2data_xfm) else None
            ),
            approx=self.inputs.approx,
            num_threads=(
                self.inputs.num_threads if isdefined(self.inputs.num_threads) else None
            ),
        ).to_filename(self._results["out_corrected"])
```